core/state/trie_prefetcher.go

```diff
@@ -266,12 +298,52 @@ func (sf *subfetcher) schedule(keys [][]byte) {
 	sf.lock.Lock()
 	sf.tasks = append(sf.tasks, keys...)
 	sf.lock.Unlock()
-
 	// Notify the prefetcher, it's fine if it's already terminated
 	select {
 	case sf.wake <- struct{}{}:
 	default:
 	}
+	atomic.AddUint32(&sf.pendingSize, uint32(len(keys)))
+}
+
+func (sf *subfetcher) scheduleParallel(keys [][]byte) {
+	var keyIndex uint32 = 0
+	childrenNum := len(sf.paraChildren)
+	if childrenNum > 0 {
+		// To feed the children first, if they are hungry.
+		// A child can handle keys with capacity of parallelTriePrefetchCapacity.
+		startIndex := len(keys) % childrenNum // randomly select the start child to avoid always feed the first one
+		for i := 0; i < childrenNum; i++ {
+			child := sf.paraChildren[startIndex]
+			startIndex = (startIndex + 1) % childrenNum
+			feedNum := parallelTriePrefetchCapacity - atomic.LoadUint32(&child.pendingSize)
```

Comment:
I am a bit worried about uint under-flow, let us protect it anyway.

Reply:
like  `if parallelTriePrefetchCapacity <=  atomic.LoadUint32(&child.pendingSize) { return }`

Reply:
Accepted, will be updated.